File: ts/lcof/22.ts
import { ListNode } from '../class'

function getKthFromEnd(head: ListNode | null, k: number): ListNode | null {
  let fast = head, slow = head, i = 0
  while (i < k) {
    fast = fast.next
    i += 1
  }
  while (fast) {
    fast = fast.next
    slow = slow.next
  }
  return slow
};

File: py3/lcof/22.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def getKthFromEnd(self, head: ListNode, k: int) -> ListNode:
        fast, slow = head, head
        cnt = 0
        while cnt < k and fast:
            fast = fast.next
            cnt += 1

        while fast:
            fast = fast.next
            slow = slow.next

        return slow